File: Components/DevExpressVCL/Demos/ExpressQuantumGrid/CBuilder/EditorsInPlaceDemo/EditorsInPlaceDemoCities.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "EditorsInPlaceDemoCities.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "cxButtons"
#pragma link "cxClasses"
#pragma link "cxControls"
#pragma link "cxCustomData"
#pragma link "cxData"
#pragma link "cxDBData"
#pragma link "cxEdit"
#pragma link "cxFilter"
#pragma link "cxGraphics"
#pragma link "cxGrid"
#pragma link "cxGridCustomTableView"
#pragma link "cxGridCustomView"
#pragma link "cxGridDBTableView"
#pragma link "cxGridLevel"
#pragma link "cxGridTableView"
#pragma link "cxLookAndFeelPainters"
#pragma link "cxStyles"
#pragma resource "*.dfm"
TEditorsInPlaceDemoCitiesForm *EditorsInPlaceDemoCitiesForm;
//---------------------------------------------------------------------------
__fastcall TEditorsInPlaceDemoCitiesForm::TEditorsInPlaceDemoCitiesForm(TComponent* Owner)
  : TForm(Owner)
{
}
//---------------------------------------------------------------------------

void __fastcall TEditorsInPlaceDemoCitiesForm::tvCitiesKeyDown(TObject *Sender, Word &Key,
    TShiftState Shift)
{
  if ((Key == VK_RETURN) && !tvCities->Controller->IsEditing)
    ModalResult = mrOk;
}
//---------------------------------------------------------------------------

void __fastcall TEditorsInPlaceDemoCitiesForm::btnAddClick(TObject *Sender)
{
  tvCities->DataController->Insert();
  tvCities->DataController->Edit();
}
//---------------------------------------------------------------------------

void __fastcall TEditorsInPlaceDemoCitiesForm::btnDeleteClick(TObject *Sender)
{
  tvCities->Controller->DeleteSelection();
}
//---------------------------------------------------------------------------

Variant TEditorsInPlaceDemoCitiesForm::GetValue()
{
  return tvCities->Controller->FocusedRecord->Values[0];
}
//---------------------------------------------------------------------------

File: Components/DevExpressVCL/Demos/ExpressQuantumGrid/Delphi/EditorsInPlaceDemo/EditorsInPlaceDemoCities.pas
unit EditorsInPlaceDemoCities;

interface

uses
{$IFDEF CLR}
  Variants,
{$ENDIF}
  Windows, Messages, Classes, Graphics, Controls, Forms,
  Dialogs, cxGridCustomTableView, cxGridTableView, cxGridDBTableView,
  StdCtrls, cxButtons, cxGridLevel, cxControls,
  cxGridCustomView, cxGrid, ExtCtrls, cxStyles, cxCustomData, cxGraphics,
  cxFilter, cxData, cxEdit, DB, cxDBData, cxClasses, EditorsInPlaceDemoData,
  cxDataStorage, cxLookAndFeelPainters;

type
  TEditorsInPlaceDemoCitiesForm = class(TForm)
    tvCities: TcxGridDBTableView;
    lvCities: TcxGridLevel;
    GridCities: TcxGrid;
    tvCitiesCity: TcxGridDBColumn;
    lbDescription: TLabel;
    btnCancel: TcxButton;
    btnOK: TcxButton;
    procedure tvCitiesKeyDown(Sender: TObject; var Key: Word;
      Shift: TShiftState);
    procedure btnAddClick(Sender: TObject);
    procedure btnDeleteClick(Sender: TObject);
  private
    function GetValue: Variant;
  public
    property Value: Variant read GetValue;
  end;

var
  EditorsInPlaceDemoCitiesForm: TEditorsInPlaceDemoCitiesForm;

implementation

{$R *.dfm}

function TEditorsInPlaceDemoCitiesForm.GetValue: Variant;
begin
  Result := tvCities.Controller.FocusedRecord.Values[0];
end;

procedure TEditorsInPlaceDemoCitiesForm.tvCitiesKeyDown(Sender: TObject;
  var Key: Word; Shift: TShiftState);
begin
  if (Key = VK_RETURN) and not tvCities.Controller.IsEditing then
    ModalResult := mrOk;
end;

procedure TEditorsInPlaceDemoCitiesForm.btnAddClick(Sender: TObject);
begin
  tvCities.DataController.Insert;
  tvCities.DataController.Edit;
end;

procedure TEditorsInPlaceDemoCitiesForm.btnDeleteClick(Sender: TObject);
begin
  tvCities.Controller.DeleteSelection;
end;

end.

